cmd/image/qcow2ova/qcow2ova.go

```diff
@@ -2,15 +2,216 @@ package qcow2ova
 	Use:   "qcow2ova",
 	Short: "Convert the qcow2 image to ova format",
-	Long:  `Convert the qcow2 image to ova format`,
+	Long: `Convert the qcow2 image to ova format
+
+Examples:
+
+  # Downloads the coreos image from remote site and converts into ova type with name rhcos-461.ova.gz
+  pvsadm image qcow2ova --image-name rhcos-461 --image-dist coreos --image-url https://mirror.openshift.com/pub/openshift-v4/ppc64le/dependencies/rhcos/4.6/4.6.1/rhcos-4.6.1-ppc64le-openstack.ppc64le.qcow2.gz
+
+  # Converts the CentOS image from the local filesystem with size 50GB
+  pvsadm image qcow2ova --image-name centos-82 --image-dist centos --image-size 50 --image-url /root/CentOS-8-GenericCloud-8.2.2004-20200611.2.ppc64le.qcow2
+
+  # Converts the RHEL image from local filesystem
+  pvsadm image qcow2ova --image-name rhel-82-29oct --image-dist rhel --rhn-user joesmith@example.com --rhn-password someValidPassword --image-url ./rhel-8.2-update-2-ppc64le-kvm.qcow2
+
+  # Converts the CentOS image from the local filesystem with OS password set
+  pvsadm image qcow2ova --image-name centos-82 --image-dist centos --os-password s0meC0mplexPassword --image-url /root/CentOS-8-GenericCloud-8.2.2004-20200611.2.ppc64le.qcow2
+`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		opt := pkg.ImageCMDOptions
+		if !utils.Contains([]string{"rhel", "centos", "coreos"}, strings.ToLower(opt.ImageDist)) {
+			klog.Errorln("--image-dist is a mandatory flag and one of these [rhel, centos, coreos]")
+			os.Exit(1)
+		}
+
+		//Read the RHNUser and RHNPassword if empty
+		if opt.ImageDist == "rhel" && (opt.RHNUser == "" || opt.RHNPassword == "") {
+			var err error
+			klog.Warning("rhn-user and rhn-password options are mandatory when image-dist is rhel, please enter the details")
+
+			//Validates and make sure input is not an empty string
+			validate := func(input string) error {
+				if len(strings.TrimSpace(input)) == 0 {
+					return fmt.Errorf("input can't be empty string")
+				}
+				return nil
+			}
+			if opt.RHNUser == "" {
+				prompt := promptui.Prompt{
+					Label:    "Enter the RHN Username",
+					Validate: validate,
+				}
+
+				opt.RHNUser, err = prompt.Run()
+				if err != nil {
+					return err
+				}
+			}
+
+			if opt.RHNPassword == "" {
+				prompt := promptui.Prompt{
+					Label:    "Enter the RHN Password",
+					Mask:     '•',
+					Validate: validate,
+				}
+
+				opt.RHNPassword, err = prompt.Run()
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		if opt.ImageDist != "coreos" && opt.OSPassword == "" {
+			var err error
+			opt.OSPassword, err = GeneratePassword(12)
+			klog.Infof("Autogenerated OS root password is: %s", opt.OSPassword)
+			if err != nil {
+				return err
+			}
+		}
+		// preflight checks validations
+		return validate.Validate()
+	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-		fmt.Println("To Be Implemented...")
+		opt := pkg.ImageCMDOptions
+
+		tmpDir, err := ioutil.TempDir("", "qcow2ova")
```

Comment:
We should add tmpdir as an optional argument. Otherwise we'll hit the previous issue of enough free space in `/tmp`

Reply:
damn! this is a complete miss from my end.! Will fix it.

Reply:
pushed a change to address this